### PR TITLE
As a Planning editor I like to use “advanced templates” for events [SDESK-6746]

### PR DESCRIPTION
**Pull request by devketanpro**

### CODE REVIEW

File: client/actions/main.ts [hunk 1]
```diff
@@ -176,12 +176,42 @@ const createNew = (itemType, item = null, updateUrl = true, modal = false) => (
...
+            },
+            news_coverage_status: {
+                qcode: coverage.news_coverage_status,
+            },
+        })),
+    }));
+}
+
 function createEventFromTemplate(template: IEventTemplate) {
+    const embeddedPlanning = template.data?.embedded_planning || [];
```
Comment (MarkLark86) [inline client/actions/main.ts:207]: This check can be done in the `getEventsAssociatedItems` function, and return empty array if not defined

**Review by MarkLark86 — CHANGES_REQUESTED** [review]

File: client/actions/events/api.ts [hunk 2]
```diff
@@ -685,7 +685,22 @@ const createEventTemplate = (itemId) => (dispatch, getState, {api, modal, notify
                 const doSave = () => {
                     api('events_template').save({
                         template_name: templateName,
-                        based_on_event: itemId,
+                        based_on_event: item._id,
+                        embedded_planning: item.associated_plannings.map((planning) => ({
```
[the remark this answers is not in the document]
Comment (MarkLark86) [reply client/actions/events/api.ts:689]: @devketanpro Please disregard this one. I'll need to do it anyways in my current task

File: server/planning/events/events_template.py [hunk 3]
```diff
@@ -31,6 +31,7 @@ class EventsTemplateResource(Resource):
     endpoint_name = "events_template"
     resource_methods = ["GET", "POST"]
     item_methods = ["GET", "DELETE", "PATCH", "PUT"]
+    allow_unknown = True
```
[the remark this answers is not in the document]
Comment (devketanpro) [reply server/planning/events/events_template.py:34]: makes sense

File: server/planning/events/events_template.py [hunk 4]
```diff
@@ -82,6 +84,9 @@ class EventsTemplateResource(Resource):
         },
         "subject": {"type": "list", "schema": {"type": "dict"}, "readonly": True},
     }
+    # add embedded_planning in the templates data schema
+    _event_fields.setdefault("embedded_planning", events_schema["embedded_planning"])
```
Comment (MarkLark86) [inline server/planning/events/events_template.py:88]: No need to use `setdefault` here, you can just place it underneath the `subject` entry above :point_up:

**Review by MarkLark86 — APPROVED** [review]

**Pull request merged**